File: 0027-remove-element/solution.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int ind = nums.length;
        for( int i=0; i<nums.length; i++){
            if( nums[i] == val){
                ind = i;
                break;
            }
        }
        
        for(int i = ind+1; i<nums.length; i++){
            if( nums[i] != val){
                //swap
                int temp = nums[ind];
                nums[ind] = nums[i];
                nums[i] = temp;
                
                ind++;
            }          
        }
        return ind;
    }
}

File: 0027-remove-element/solution.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        // look for the 1st element != val;
        
        int j=nums.size();  // vvi vvi vvi
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] == val){
                j = i;
                break;
            }
        }

        for(int i=j+1; i<nums.size(); i++){
            if(nums[i] != val){
                swap(nums[i], nums[j]);
                j++;
            }
        }

        return j;
    }
};
